### DISCUSSION TITLE
How to use callback function of NextAuth for the client API?

### DISCUSSION THREAD

**Discussion by anand-dehru** [nextauthjs/next-auth#8474] [Help]
### Question 💬

The scenario is I want role property  in the respionse of the person which has logged in. But as we know the session data returned to the client does not contain sensitive information such as the Session Token. So we need callbacks to get additional data in the client side. I tried to use callback but still it is showing role does not exist in the "user".

This is my backend code

`import { PrismaAdapter } from "@next-auth/prisma-adapter";
import bcrypt from "bcrypt";
import NextAuth, { AuthOptions } from "next-auth";
import client from "@/lib/prismadb";
import CredentialsProvider from "next-auth/providers/credentials";
import GoogleProvider from "next-auth/providers/google";

export const authOptions: AuthOptions = {
  adapter: PrismaAdapter(client),
  providers: [
    GoogleProvider({
      clientId: process.env.GOOGLE_CLIENT_ID as string,
      clientSecret: process.env.GOOGLE_CLIENT_SECRET as string,
    }),
    CredentialsProvider({
      name: "credentials",
      credentials: {
        email: { label: "email", type: "text" },
        password: { label: "password", type: "password" },
      },
      async authorize(credentials) {
        // check input
        if (!credentials?.email || !credentials?.password)
          throw new Error("Invalid credentials");

        // check user exists
        const user = await client.user.findUnique({
          where: {
            email: credentials.email,
          },
        });

        if (!user || !user.hashedPassword)
          throw new Error("Invalid credentials");

        // check password
        const passMatch = await bcrypt.compare(
          credentials.password,
          user.hashedPassword
        );

        if (!passMatch) throw new Error("Incorrect password");
         return user;
      },
    }),
  ],
  debug: process.env.NODE_ENV === "development",
  session: {
    strategy: "jwt",
  },
 callback:{
?????????????? 
},
  pages: {
    signIn: "/signin",
  },
  secret: process.env.NEXTAUTH_SECRET,
};

const handler = NextAuth(authOptions);

export { handler as GET, handler as POST };
`

this is my front end code 

const session = useSession();
  console.log(session, "+++");
  signIn("credentials", {
        ...data,
        redirect: false,
      })
        .then((callback) => {
          console.log(callback, "call");
          if (callback?.error) toast.error("Invalid credentials");
          if (callback?.ok && !callback?.error) {
            toast.success("Logged In");
            router.push("/");
          }
        })
        .finally(() => setIsLoading(false));

I am just getting name,email and image in the response but not the role so how to do that?

### How to reproduce ☕️

.

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by coded58** [comment]
check out this... it assisted me in configuring this same issue https://www.youtube.com/watch?v=khNwrFJ-Xqs&list=WL&index=21
